Zend/zend_compile.c

```diff
@@ -221,6 +221,19 @@ static const builtin_type_info builtin_types[] = {
 	{NULL, 0, IS_UNDEF}
 };
 
+typedef struct {
+	const char *name;
+	size_t name_len;
+	const char *correct_name;
+} confusable_type_info;
+
+static const confusable_type_info confusable_types[] = {
+	{ZEND_STRL("boolean"), "bool"},
+	{ZEND_STRL("integer"), "int"},
+	{ZEND_STRL("double"), "float"},
+	{ZEND_STRL("resource"), NULL},
```

Comment:
Might suggest also:

```
{ZEND_STRL("stream"), NULL},
```

Since `get_resource_type(fopen(...))` returns `stream` and there's been at least one confusion on this https://stackoverflow.com/q/40637939/2908724.

Though I realize this might spiral into all the one-word resource types, but we are just providing guidance here.

Reply:
Personally I'd prefer not to include specific resource types here. For things like `stream` it gets increasingly likely that this is a legitimate class name and not an attempt to use a resource type.